### ISSUE TITLE
meryl fails to detemine memory and segments

### ISSUE THREAD

**Issue by brianwalenz** [marbl/canu#1206]
Same here. Traced the problem back to `Meryl.pm`, where in this line
```
    print STDERR "--  Will count kmers using $merylSegments jobs, each using $merylMemory GB and $thr threads.\n";
```
both `$merylMemory` and `$merylSegments` are still undefined.
This leads to `memory=` without value when the `meryl-count.sh` is generated, probably from here:
```
        print F "$bin/meryl -C k=$merSize threads=$thr memory=$mem \\\n";
```
Will give 1.7 a try, thanks for mentioning that it works.

_Originally posted by @spock in https://github.com/marbl/canu/issues/1138#issuecomment-451969944_

**Comment by brianwalenz** [comment]
The original issue is for failures with PBSpro.  This seems unrelated, so I moved it to a new issue.

What is canu reporting to the screen?

Can you successfully run, by hand, any of the commands in meryl-configure.sh?  On an ecoli, I get:
```
> meryl -C k=22 threads=4 memory=12 count segment=1/01 ../../test.seqStore

Counting 168  million canonical 22-mers from 1 input file:
    canu-seqStore: ../../test.seqStore


SIMPLE MODE
-----------

  Disabled for mers larger than 20.


COMPLEX MODE
------------

prefix     # of   struct   kmers/    segs/     data    total
  bits   prefix   memory   prefix   prefix   memory   memory
------  -------  -------  -------  -------  -------  -------
     1     2  P   113 kB    84 MM  7238  S   904 MB   904 MB
     2     4  P   110 kB    42 MM  3535  S   883 MB   883 MB
     3     8  P   108 kB    21 MM  1726  S   863 MB   863 MB
     4    16  P   107 kB    10 MM   842  S   842 MB   842 MB
     5    32  P   106 kB  5385 kM   411  S   822 MB   822 MB
     6    64  P   107 kB  2692 kM   200  S   800 MB   800 MB
     7   128  P   112 kB  1346 kM    98  S   784 MB   784 MB
     8   256  P   124 kB   673 kM    48  S   768 MB   768 MB
     9   512  P   152 kB   336 kM    24  S   768 MB   768 MB
    10  1024  P   208 kB   168 kM    12  S   768 MB   768 MB  Best Value!
    11  2048  P   320 kB    84 kM     6  S   768 MB   768 MB
    12  4096  P   544 kB    42 kM     3  S   768 MB   768 MB
    13  8192  P  1024 kB    21 kM     2  S  1024 MB  1025 MB
    14    16 kP  1920 kB    10 kM     1  S  1024 MB  1025 MB
    15    32 kP  3840 kB  5386  M     1  S  2048 MB  2051 MB
    16    64 kP  7680 kB  2693  M     1  S  4096 MB  4103 MB
    17   128 kP    15 MB  1347  M     1  S  8192 MB  8207 MB
    18   256 kP    30 MB   674  M     1  S    16 GB    16 GB


FINAL CONFIGURATION
-------------------

Configured complex mode for 0.750 GB memory per batch, and up to 1 batch.

Bye.
```

**Comment by spock** [comment]
I'm also on PBSPro:
```
-- Detected PBSPro '18.2.1.20180622093321' with 'pbsnodes' binary in /opt/pbs/bin/pbsnodes.
-- Detecting PBSPro resources.
```

**Comment by brianwalenz** [comment]
Potentially fixed in e13467a0ada171b9f70dd9ea615452cd707ea0ac and df2dbf1df0fa8fd0a98a6f281dafcb131f57dc64.
